### PR TITLE
Improve handling of precursor m/z when reading/writing mzML

### PR DESCRIPTION
**Pull request by hendrikweisser**
This addresses issue #4168 using the approach mentioned here: https://github.com/OpenMS/OpenMS/issues/4168#issuecomment-508520131

What remains is checking for the meta value "isolation window target m/z" in the isolation window visualisation code for TOPPView. Can anyone point me to where this should be changed?

### CODE REVIEW

File: src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp [hunk 1]
```diff
@@ -3836,7 +3853,7 @@ namespace OpenMS
         os << "\t\t\t\t\t\t\t<cvParam cvRef=\"MS\" accession=\"MS:1000044\" name=\"dissociation method\" />\n";
       }
       //as "precursor" has no own user param its userParam is stored here
-      writeUserParam_(os, precursor, 7, "/mzML/run/spectrumList/spectrum/precursorList/precursor/activation/cvParam/@accession", validator);
+      writeUserParam_(os, precursor, 7, "/mzML/run/spectrumList/spectrum/precursorList/precursor/activation/cvParam/@accession", validator, {"isolation window target m/z", "selected ion m/z"});
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp]: please add some extra documentation why these are not written out

File: src/tests/topp/FileMerger_6_output.mzML [hunk 2]
```diff
@@ -199,9 +199,6 @@
 				</scanList>
 				<precursorList count="1">
 					<precursor>
-						<isolationWindow>
-							<cvParam cvRef="MS" accession="MS:1000827" name="isolation window target m/z" value="445.34" unitAccession="MS:1000040" unitName="m/z" unitCvRef="MS" />
-						</isolationWindow>
 						<selectedIonList count="1">
```
Comment (hroest) [inline src/tests/topp/FileMerger_6_output.mzML]: @hendrikweisser is this a good idea to not write this out any more?

Comment (hendrikweisser) [reply src/tests/topp/FileMerger_6_output.mzML]: Yes. The element actually isn't present in the input file that gets merged, so the new behaviour is correct.

File: src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp (hunk 1 above)
Comment (hendrikweisser) [reply src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp]: Done.

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
Do we have any experience if the stripped information is used by third-party tools?

**Comment by hendrikweisser** [comment]
@timosachsenberg:
What do you mean by "stripped information"?

**Comment by timosachsenberg** [comment]
If some thid-party tools rely on these meta values being present.

**Comment by hendrikweisser** [comment]
I'm still not sure I understand. There would have to be a meta value with one of the names used internally for it to be misinterpreted and removed. Possible in principle, but seems unlikely, and in any case an external tool (that accepts "standard" mzML) could not rely on such a meta value being present.

**Comment by timosachsenberg** [comment]
ok can you merge against develop. Would like to get this included for testing in the release candidate.

**Comment by hendrikweisser** [comment]
Can do. Which option should I use (normal, squash, rebase)?

Do you know where changes would need to be made in TOPPView to fix the isolation window visualisation?

**Comment by timosachsenberg** [comment]
should be in https://github.com/OpenMS/OpenMS/blob/develop/src/openms_gui/source/VISUAL/TOPPViewIdentificationViewBehavior.cpp#L827
you can just merge and we can squash and merge later

**Comment by hendrikweisser** [comment]
Thanks for the TOPPView pointer, Timo. I've made the small change and it seems to have the desired effect - see screenshot (left: before, right: after).

TOPPView_precursor_fix

Still okay to merge?

**Comment by timosachsenberg** [comment]
I think we can merge and see how/if it negatively affects the visualization of other data.

**Pull request merged**